go/runtime/registry/config.go

```diff
@@ -201,6 +202,18 @@ func newConfig(dataDir string, commonStore *persistent.CommonStore, consensus co
 			return nil, fmt.Errorf("unsupported runtime provisioner: %s", p)
 		}
 
+		// Configure optional load balancing.
+		if numInstances := config.GlobalConfig.Runtime.LoadBalancer.NumInstances; numInstances > 1 {
```

Comment:
Since the load balance constructor returns inner provisioner, I would always enable the balancer. And change the constructor to ignore 0 and 1 (or maybe config to `extra_instances`). As a consequence, the constructor won't return an error anymore.

```go
		// Configure load balancing.
		for tee, rp := range rh.Provisioners {
			rh.Provisioners[tee] = hostLoadBalance.New(rp, config.GlobalConfig.Runtime.LoadBalancer)
		}
```